public-ui/src/app/data-browser/chart/chart.component.ts

```diff
@@ -233,10 +275,10 @@ export class ChartComponent implements OnChanges {
     // console.log('Global series clicked ', this.analysis, 'Clicked analysis', event.point);
   }
 
-  public makeCountChartOptions() {
+  public makeCountChartOptions(results: any, analysisName: string) {
     let data = [];
     let cats = [];
-    for (const a of this.analysis.results) {
+    for (const a of results) {
```

Comment:
use 'let' instead of 'const' when assigning a variable a value that gets reassigned.